winbuild.bat

```diff
@@ -0,0 +1,7 @@
+rmdir /S /Q x64
+call phpize
+call configure --enable-scoutapm --enable-debug --with-php-build="C:\php-sdk\phpdev\vs16\x64\deps" --with-prefix="C:\php\"
+call sed -i 's/-d extension=/-d zend_extension=/g' Makefile
+call sed -i 's/CC="$(PHP_CL)"/CC="cl.exe"/g' Makefile
```

Comment:
Is this really necessary? The build works for me with PHP 7.4 and 8.2 without this.

Reply:
With my setup, I could not understand why, despite `PHP_CL` being set to `cl.exe`, the `CC` variable was just an empty string, causing the builds to fail :/ could not understand why:

```
$ winbuild.bat

Z:\
$ rmdir /S /Q x64

Z:\
$ call phpize
Rebuilding configure.js
C:\php-sdk\php-8.1.7-devel-vs16-x64
module ...
Now run 'configure --help'
Saving configure options to config.nice.bat
Checking for cl.exe ...  <in default path>
  Detected compiler Visual C++ 2019
  Detected 64-bit compiler
Checking for link.exe ...  <in default path>
Checking for nmake.exe ...  <in default path>
Checking for lib.exe ...  <in default path>
Checking for bison.exe ...  <in default path>
  Detected bison version 3.3.2
Checking for sed.exe ...  <in default path>
Checking for re2c.exe ...  <in default path>
  Detected re2c version 1.1.1
Checking for zip.exe ...  <in default path>
Checking for lemon.exe ...  <in default path>
Checking for 7za.exe ...  <in default path>
Checking for mc.exe ...  C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64
Checking for mt.exe ...  C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64

Build dir: Z:\\x64\Release_TS
PHP Core:  php8ts.dll and php8ts.lib

Enabling extension Z:\ [shared]

Creating build dirs...
Generating files...
Generating Makefile
Generating main/config.pickle.h
Generating pickle deps
Done.



Enabled extensions:
--------------------
| Extension | Mode |
--------------------
--------------------


Enabled Zend extensions:
----------------------
| Extension | Mode   |
----------------------
| scoutapm  | shared |
----------------------


---------------------------------------
|                   |                 |
---------------------------------------
| Build type        | Release         |
| Thread Safety     | Yes             |
| Compiler          | Visual C++ 2019 |
| Architecture      | x64             |
| Optimization      | PGO disabled    |
| Native intrinsics | SSE2            |
| Static analyzer   | disabled        |
---------------------------------------


Type 'nmake' to build PHP

Microsoft (R) Program Maintenance Utility Version 14.29.30145.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        "" /D WIN32_ONLY_COMPILER=1 /wd4005 /D ZEND_COMPILE_DL_EXT=1 /D COMPILE_DL_SCOUTAPM /D SCOUTAPM_EXPORTS=1 /nologo /I C:\php-sdk\php-8.1.7-devel-vs16-x64/include /I C:\php-sdk\php-8.1.7-devel-vs16-x64/include/main /I C:\php-sdk\php-8.1.7-devel-vs16-x64/include/Zend /I C:\php-sdk\php-8.1.7-devel-vs16-x64/include/TSRM /I C:\php-sdk\php-8.1.7-devel-vs16-x64/include/ext /D _WINDOWS /D WINDOWS=1 /D ZEND_WIN32=1 /D PHP_WIN32=1 /D WIN32 /D _MBCS /D _USE_MATH_DEFINES /FD /wd4996 /Qspectre /guard:cf /Zc:inline /Zc:__cplusplus /d2FuncCache1 /Zc:wchar_t /MP /LD /MD /Ox /D NDebug /D NDEBUG /GF /D ZEND_DEBUG=0 /D ZTS=1 /I "C:\php-sdk\phpdev\vs16\x64\deps\include" /FoZ:\\x64\Release_TS\\ /FpZ:\\x64\Release_TS\\ /FRZ:\\x64\Release_TS\\ /FdZ:\\x64\Release_TS\\ /c Z:\\scout_curl_wrapper.c Z:\\scout_execute_ex.c Z:\\scout_file_wrapper.c Z:\\scout_functions.c Z:\\scout_internal_handlers.c Z:\\scout_observer.c Z:\\scout_pdo_wrapper.c Z:\\scout_recording.c Z:\\scout_utils.c Z:\\zend_scoutapm.c
The system cannot find the path specified.
NMAKE : fatal error U1077: '"' : return code '0x1'
Stop.

Microsoft (R) Program Maintenance Utility Version 14.29.30145.0
Copyright (C) Microsoft Corporation.  All rights reserved.

          "C:\php\\php.exe" -n -d zend_extension=Z:\\x64\Release_TS\\php_scoutapm.dll -m
Failed loading Z:\\x64\Release_TS\\php_scoutapm.dll
[PHP Modules]
bcmath
calendar
Core
ctype
date
dom
filter
hash
iconv
json
libxml
mysqlnd
pcre
PDO
Phar
readline
Reflection
session
SimpleXML
SPL
standard
tokenizer
xml
xmlreader
xmlwriter
zip
zlib

[Zend Modules]


Z:\
$
```

Reply:
> With my setup, I could not understand why, despite `PHP_CL` being set to `cl.exe`, the `CC` variable was just an empty string,

That is very strange, indeed. If there is an environment variable `CC` that would only override the definition in the Makefile if you call `nmake /e`, but apparently your not passing the `/e` option. And apparently you also don't define CC when calling `nmake` (e.g. `nmake CC=`), so have no idea what is going there. But at least instead of modifying the Makefile, you could probably do `nmake CC=cl.exe` (which feels slightly cleaner).

Reply:
> But at least instead of modifying the Makefile, you could probably do `nmake CC=cl.exe` (which feels slightly cleaner).

Thanks - updated to use this, yeah tis slightly better.